src/mongo_logger.ts

```diff
@@ -119,61 +189,273 @@ function parseSeverityFromString(s?: string): SeverityLevel | null {
+      log = attachCommandFields(log, logObject);
+      log.message = 'Command succeeded';
+      log.durationMS = logObject.duration;
+      log.reply = EJSON.stringify(logObject.reply);
+      break;
+    case COMMAND_FAILED:
+      log = attachCommandFields(log, logObject);
+      log.message = 'Command failed';
+      log.durationMS = logObject.duration;
+      log.failure = logObject.failure;
+      break;
+    case CONNECTION_POOL_CREATED:
+      log = attachConnectionFields(log, logObject);
+      log.message = 'Connection pool created';
+      if (logObject.options) {
+        const { maxIdleTimeMS, minPoolSize, maxPoolSize, maxConnecting, waitQueueTimeoutMS } =
+          logObject.options;
+        log = {
+          ...log,
+          maxIdleTimeMS,
+          minPoolSize,
+          maxPoolSize,
+          maxConnecting,
+          waitQueueTimeoutMS
+        };
+        log.waitQueueSize = logObject.waitQueueSize;
+      }
+      break;
+    case CONNECTION_POOL_READY:
+      log = attachConnectionFields(log, logObject);
+      log.message = 'Connection pool ready';
+      break;
+    case CONNECTION_POOL_CLEARED:
+      log = attachConnectionFields(log, logObject);
+      log.message = 'Connection pool cleared';
+      if (logObject.serviceId?._bsontype === 'ObjectId') {
+        log.serviceId = logObject.serviceId.toHexString();
+      }
+      break;
+    case CONNECTION_POOL_CLOSED:
+      log = attachConnectionFields(log, logObject);
+      log.message = 'Connection pool closed';
+      break;
+    case CONNECTION_CREATED:
+      log = attachConnectionFields(log, logObject);
+      log.message = 'Connection created';
+      if ('connectionId' in logObject) {
+        log.driverConnectionId = logObject.connectionId;
+      }
+      break;
+    case CONNECTION_READY:
+      log = attachConnectionFields(log, logObject);
+      log.message = 'Connection ready';
+      if ('connectionId' in logObject) {
+        log.driverConnectionId = logObject.connectionId;
+      }
+      break;
+    case CONNECTION_CLOSED:
+      log = attachConnectionFields(log, logObject);
+      log.message = 'Connection closed';
+      if ('connectionId' in logObject) {
+        log.driverConnectionId = logObject.connectionId;
+      }
+      switch (logObject.reason) {
+        case 'stale':
+          log.reason = 'Connection became stale because the pool was cleared';
+          break;
+        case 'idle':
+          log.reason =
+            'Connection has been available but unused for longer than the configured max idle time';
+          break;
+        case 'error':
+          log.reason = 'An error occurred while using the connection';
+          if (logObject.error) {
+            log.error = logObject.error;
+          }
+          break;
+        case 'poolClosed':
+          log.reason = 'Connection pool was closed';
+          break;
+        default:
+        // Omit if we have some other reason as it would be invalid
```

Comment:
MongoRuntimeError? or log.reason == `Unknown close reason: ${logObject.reason}`?  Kinda prefer the latter